common/messages/heartbeat.go

```diff
@@ -326,7 +327,15 @@ func (m *Heartbeat) FollowerExecute(is interfaces.IState) {
 		if auditServer.GetChainID().IsSameAs(m.IdentityChainID) {
 			if m.IdentityChainID.IsSameAs(is.GetIdentityChainID()) {
 				if m.SecretNumber != is.GetSalt(m.Timestamp) {
-					panic("We have seen a heartbeat using our Identity that isn't ours")
+					lLeaderHeight := is.GetLLeaderHeight()
+					if m.DBHeight == lLeaderHeight {
```

Comment:
I think this change is valid, however, dependent holding is finalized and will be in parchment. With that, we can hold this message until we hit the height it is for. This would remove this state logic from this msg execute, which is the direction we are headed with the refactor. When the dependent holding is apart of the new parchment head, I say we move this.

Reply:
Until this is resolved in dependent holding, this check is valid :+1: